### ISSUE TITLE
(Traefik v2.2) Unable to obtain ACME certificate with DNS challenge using Go Daddy 

### ISSUE THREAD

**Issue by floatingpurr** [containous/traefik#6686]
### Do you want to request a *feature* or report a *bug*?

A potential Bug

### What did you do?

I've just posted this issue in the community forum, I try to report it also here, just to be sure this is not a bug.

I'm simply trying to set up a wildcard certificate with DNS challenge using Go Daddy. This is my compose file:

```yml
version: '3.7'

services:
  traefik:
    image: traefik:v2.2
    container_name: traefik
    restart: always
    env_file:
      - .provider.env
      # .provider.env contains `GODADDY_API_KEY` and `GODADDY_API_SECRET`
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./tls-certificates:/tls-certificates
    ports:
      # http
      - 8080:80
      # https
      - 443:443
    command:
      - --api.dashboard=true
      - --providers.docker=true
      - --providers.docker.exposedbydefault=false
      - --providers.docker.network=proxy
      - --entrypoints.webinsecure.address=:80
      - --entrypoints.websecure.address=:443

      # --certificatesresolvers.<name> Certificates resolvers configuration
      # ACME V2 supports wildcard certificates.
      # Wildcard certificates can only be generated through a DNS-01 challenge.
      - --certificatesresolvers.wildcard-godaddy.acme.dnschallenge=true
      - --certificatesResolvers.wildcard-godaddy.acme.dnsChallenge.provider=godaddy
      - --certificatesResolvers.wildcard-godaddy.acme.dnsChallenge.delayBeforeCheck=0
      # Email address used for registration.
      - --certificatesresolvers.wildcard-godaddy.acme.email=foo@example.org
      # Certificates storage
      - --certificatesresolvers.wildcard-godaddy.acme.storage=/tls-certificates/acme.json

    networks:
      - proxy

    labels:

      - "traefik.enable=true"
      - "traefik.http.routers.traefik.entrypoints=webinsecure"
      - "traefik.http.routers.traefik.rule=Host(`traefik.example.org`)"

      - "traefik.http.middlewares.traefik-auth.basicauth.users=${DASHBOARD_USERNAME}:${DASHBOARD_PASSWORD}"
      - "traefik.http.middlewares.traefik-https-redirect.redirectscheme.scheme=https"
      - "traefik.http.routers.traefik.middlewares=traefik-https-redirect"

      - "traefik.http.routers.traefik-secure.entrypoints=websecure"
      - "traefik.http.routers.traefik-secure.rule=Host(`traefik.example.org`)"
      - "traefik.http.routers.traefik-secure.middlewares=traefik-auth"
      - "traefik.http.routers.traefik-secure.tls=true"
      - "traefik.http.routers.traefik-secure.tls.certresolver=wildcard-godaddy"
      - "traefik.http.routers.traefik-secure.tls.domains[0].main=example.org"
      - "traefik.http.routers.traefik-secure.tls.domains[0].sans=*.example.org"
      - "traefik.http.routers.traefik-secure.service=api@internal"


networks:
  proxy:
    external: true
```

I generated an API key into the GoDaddy management panel. In the dns, I've the entry `A`,`*` pointing to the IP address where traefik runs.

Before this, I generated in the past a certificate for my domain with the "classic" `acme.tlschallenge` but I believe it's not the point.

### What did you expect to see?

A correct wildcard certificate generated for the pattern `*.example.org`

### What did you see instead?

I get: Unable to obtain ACME certificate for domains \"example.org,*.example.org\"

> traefik    | time="2020-04-16T10:24:59Z" level=error msg="Unable to obtain ACME certificate for domains \"example.org,*.example.org\" : unable to generate a certificate for the domains [example.org *.example.org]: acme: Error -> One or more domains had a problem:\n[example.org] acme: error: 403 :: urn:ietf:params:acme:error:unauthorized :: During secondary validation: Incorrect TXT record \"null\" found at _acme-challenge.example.org, url: \n" providerName=wildcard-godaddy.acme

### Output of `traefik version`: (_What version of Traefik are you using?_)

**`Traefik:2.2`**

```
Version:      2.2.0
Codename:     chevrotin
Go version:   go1.14.1
Built:        2020-03-25T17:32:57Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

OS: Ubuntu 18.04

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
Starting traefik ... done
Attaching to traefik
traefik    | time="2020-04-16T10:23:02Z" level=info msg="Configuration loaded from flags."
traefik    | time="2020-04-16T10:23:02Z" level=info msg="Traefik version 2.2.0 built on 2020-03-25T17:32:57Z"
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Static configuration loaded {\"global\":{\"checkNewVersion\":true},\"serversTransport\":{\"maxIdleConnsPerHost\":200},\"entryPoints\":{\"webinsecure\":{\"address\":\":80\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":10000000000},\"respondingTimeouts\":{\"idleTimeout\":180000000000}},\"forwardedHeaders\":{},\"http\":{}},\"websecure\":{\"address\":\":443\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":10000000000},\"respondingTimeouts\":{\"idleTimeout\":180000000000}},\"forwardedHeaders\":{},\"http\":{}}},\"providers\":{\"providersThrottleDuration\":2000000000,\"docker\":{\"watch\":true,\"endpoint\":\"unix:///var/run/docker.sock\",\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"network\":\"proxy\",\"swarmModeRefreshSeconds\":15000000000}},\"api\":{\"dashboard\":true},\"log\":{\"level\":\"DEBUG\",\"format\":\"common\"},\"certificatesResolvers\":{\"wildcard-godaddy\":{\"acme\":{\"email\":\"foo@example.org\",\"caServer\":\"https://acme-v02.api.letsencrypt.org/directory\",\"storage\":\"/tls-certificates/acme.json\",\"keyType\":\"RSA4096\",\"dnsChallenge\":{\"provider\":\"godaddy\"}}}}}"
traefik    | time="2020-04-16T10:23:02Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/contributing/data-collection/\n"
traefik    | time="2020-04-16T10:23:02Z" level=info msg="Starting provider aggregator.ProviderAggregator {}"
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Start TCP Server" entryPointName=websecure
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Start TCP Server" entryPointName=webinsecure
traefik    | time="2020-04-16T10:23:02Z" level=info msg="Starting provider *acme.Provider {\"email\":\"foo@example.org\",\"caServer\":\"https://acme-v02.api.letsencrypt.org/directory\",\"storage\":\"/tls-certificates/acme.json\",\"keyType\":\"RSA4096\",\"dnsChallenge\":{\"provider\":\"godaddy\"},\"ResolverName\":\"wildcard-godaddy\",\"store\":{},\"ChallengeStore\":{}}"
traefik    | time="2020-04-16T10:23:02Z" level=info msg="Testing certificate renew..." providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:02Z" level=info msg="Starting provider *docker.Provider {\"watch\":true,\"endpoint\":\"unix:///var/run/docker.sock\",\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"network\":\"proxy\",\"swarmModeRefreshSeconds\":15000000000}"
traefik    | time="2020-04-16T10:23:02Z" level=info msg="Starting provider *traefik.Provider {}"
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Configuration received from provider wildcard-godaddy.acme: {\"http\":{},\"tls\":{}}" providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Configuration received from provider internal: {\"http\":{\"services\":{\"api\":{},\"dashboard\":{},\"noop\":{}}},\"tcp\":{},\"tls\":{}}" providerName=internal
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="No default certificate, generating one"
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Provider connection established with docker 19.03.8 (API 1.40)" providerName=docker
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Filtering disabled container" providerName=docker container=app-covid-19-ita-bot-57b7faa41c5c63320962127cadb3bf2fcc4af3a3e5c662468f73e11a9f8dc668
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Filtering disabled container" providerName=docker container=mongo-covid-19-ita-bot-1930b8dd265cda9a0ba45da731df8ae740ccea85f3eb3d8e934e63316b92ae55
traefik    | time="2020-04-16T10:23:02Z" level=debug msg="Configuration received from provider docker: {\"http\":{\"routers\":{\"traefik\":{\"entryPoints\":[\"webinsecure\"],\"middlewares\":[\"traefik-https-redirect\"],\"service\":\"traefik-traefik\",\"rule\":\"Host(`traefik.example.org`)\"},\"traefik-secure\":{\"entryPoints\":[\"websecure\"],\"middlewares\":[\"traefik-auth\"],\"service\":\"api@internal\",\"rule\":\"Host(`traefik.example.org`)\",\"tls\":{\"certResolver\":\"wildcard-godaddy\",\"domains\":[{\"main\":\"example.org\",\"sans\":[\"*.example.org\"]}]}}},\"services\":{\"traefik-traefik\":{\"loadBalancer\":{\"servers\":[{\"url\":\"http://192.168.48.2:80\"}],\"passHostHeader\":true}}},\"middlewares\":{\"traefik-auth\":{\"basicAuth\":{\"users\":[\"superciccio14:$apr1$trpV7fFr$0wwX2dOyOAXkcHYR41E1Y0\"]}},\"traefik-https-redirect\":{\"redirectScheme\":{\"scheme\":\"https\"}}}},\"tcp\":{},\"udp\":{}}" providerName=docker
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="No default certificate, generating one"
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Creating middleware" routerName=traefik@docker serviceName=traefik-traefik middlewareName=pipelining middlewareType=Pipelining entryPointName=webinsecure
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Creating load-balancer" entryPointName=webinsecure routerName=traefik@docker serviceName=traefik-traefik
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Creating server 0 http://192.168.48.2:80" routerName=traefik@docker serviceName=traefik-traefik serverName=0 entryPointName=webinsecure
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Added outgoing tracing middleware traefik-traefik" middlewareName=tracing middlewareType=TracingForwarder entryPointName=webinsecure routerName=traefik@docker
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Creating middleware" middlewareType=RedirectScheme entryPointName=webinsecure routerName=traefik@docker middlewareName=traefik-https-redirect@docker
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Setting up redirection to https " middlewareName=traefik-https-redirect@docker middlewareType=RedirectScheme entryPointName=webinsecure routerName=traefik@docker
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Adding tracing to middleware" middlewareName=traefik-https-redirect@docker entryPointName=webinsecure routerName=traefik@docker
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Creating middleware" entryPointName=webinsecure middlewareName=traefik-internal-recovery middlewareType=Recovery
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Added outgoing tracing middleware api@internal" middlewareType=TracingForwarder entryPointName=websecure routerName=traefik-secure@docker middlewareName=tracing
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Creating middleware" middlewareName=traefik-auth@docker middlewareType=BasicAuth entryPointName=websecure routerName=traefik-secure@docker
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Adding tracing to middleware" routerName=traefik-secure@docker middlewareName=traefik-auth@docker entryPointName=websecure
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="Creating middleware" middlewareType=Recovery entryPointName=websecure middlewareName=traefik-internal-recovery
traefik    | time="2020-04-16T10:23:03Z" level=debug msg="No default certificate, generating one"
traefik    | time="2020-04-16T10:23:04Z" level=debug msg="Looking for provided certificate(s) to validate [\"example.org\" \"*.example.org\"]..." providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:04Z" level=debug msg="Domains [\"example.org\" \"*.example.org\"] need ACME certificates generation for domains \"example.org,*.example.org\"." providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:04Z" level=debug msg="Loading ACME certificates [example.org *.example.org]..." providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:09Z" level=debug msg="Building ACME client..." providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:09Z" level=debug msg="https://acme-v02.api.letsencrypt.org/directory" providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:09Z" level=info msg=Register... providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:09Z" level=debug msg="legolog: [INFO] acme: Registering account for foo@example.org"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="Using DNS Challenge provider: godaddy" providerName=wildcard-godaddy.acme
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [example.org, *.example.org] acme: Obtaining bundled SAN certificate"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [*.example.org] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/<id1>"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [example.org] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/<id2>"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [*.example.org] acme: use dns-01 solver"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [example.org] acme: Could not find solver for: tls-alpn-01"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [example.org] acme: Could not find solver for: http-01"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [example.org] acme: use dns-01 solver"
traefik    | time="2020-04-16T10:23:10Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Preparing to solve DNS-01"
traefik    | time="2020-04-16T10:23:11Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Trying to solve DNS-01"
traefik    | time="2020-04-16T10:23:11Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Checking DNS record propagation using [127.0.0.11:53]"
traefik    | time="2020-04-16T10:23:11Z" level=debug msg="legolog: [INFO] Wait for propagation [timeout: 2m0s, interval: 2s]"
traefik    | time="2020-04-16T10:23:11Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:13Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:15Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:17Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:19Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:21Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:23Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:25Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:27Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:29Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:31Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:33Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:23:43Z" level=debug msg="legolog: [INFO] [*.example.org] The server validated our request"
traefik    | time="2020-04-16T10:23:43Z" level=debug msg="legolog: [INFO] [*.example.org] acme: Cleaning DNS-01 challenge"
traefik    | time="2020-04-16T10:23:44Z" level=debug msg="legolog: [INFO] sequence: wait for 1m0s"
traefik    | time="2020-04-16T10:24:44Z" level=debug msg="legolog: [INFO] [example.org] acme: Preparing to solve DNS-01"
traefik    | time="2020-04-16T10:24:45Z" level=debug msg="legolog: [INFO] [example.org] acme: Trying to solve DNS-01"
traefik    | time="2020-04-16T10:24:45Z" level=debug msg="legolog: [INFO] [example.org] acme: Checking DNS record propagation using [127.0.0.11:53]"
traefik    | time="2020-04-16T10:24:45Z" level=debug msg="legolog: [INFO] Wait for propagation [timeout: 2m0s, interval: 2s]"
traefik    | time="2020-04-16T10:24:45Z" level=debug msg="legolog: [INFO] [example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:24:47Z" level=debug msg="legolog: [INFO] [example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:24:49Z" level=debug msg="legolog: [INFO] [example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:24:51Z" level=debug msg="legolog: [INFO] [example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:24:53Z" level=debug msg="legolog: [INFO] [example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:24:55Z" level=debug msg="legolog: [INFO] [example.org] acme: Waiting for DNS record propagation."
traefik    | time="2020-04-16T10:24:58Z" level=debug msg="legolog: [INFO] [example.org] acme: Cleaning DNS-01 challenge"
traefik    | time="2020-04-16T10:24:58Z" level=debug msg="legolog: [INFO] Skipping deactivating of valid auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/<id1>"
traefik    | time="2020-04-16T10:24:58Z" level=debug msg="legolog: [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/<id2>"
traefik    | time="2020-04-16T10:24:59Z" level=debug msg="legolog: [INFO] Unable to deactivate the authorization: https://acme-v02.api.letsencrypt.org/acme/authz-v3/<id2>"
traefik    | time="2020-04-16T10:24:59Z" level=error msg="Unable to obtain ACME certificate for domains \"example.org,*.example.org\" : unable to generate a certificate for the domains [example.org *.example.org]: acme: Error -> One or more domains had a problem:\n[example.org] acme: error: 403 :: urn:ietf:params:acme:error:unauthorized :: During secondary validation: Incorrect TXT record \"null\" found at _acme-challenge.example.org, url: \n" providerName=wildcard-godaddy.acme
```

Not sure whether this is a bug or another problem w/ GoDaddy. Sorry if this issue looks trivial but I'm completely puzzled.

**Comment by ldez** [comment]
Hello,

I don't think that is a bug with Traefik, because it's impossible for Traefik to create a `null` TXT record.

The support will continue on https://community.containo.us/t/unable-to-obtain-acme-certificate-for-wildcard-domains/5408

**Comment by floatingpurr** [comment]
Follow-up. Just found this similar issue #5277. Also in my case, looks like it is

> a timing problem with the request and the subsequent "clearing" of same request.

Also in my case

> I've cleaned the acme.json a couple of times, as well as deleting the txt record at godaddy.
> 
> Watching while it runs, I can see txt records appear, before finally being replaced by a "null" value at the end (this seems to be by design as you can't delete txt records thru godaddys API) 

In that case, the OP fixed using:

```
      - "traefik.http.routers.traefik-secure.tls.domains[0].main=*.example.org"
      - "traefik.http.routers.traefik-secure.tls.domains[0].sans=example.org"
```

In my specific case, that did not change the error I'm experiencing.

**Comment by floatingpurr** [comment]
Ok @ide

> Hello,
> 
> I don't think that is a bug with Traefik, because it's impossible for Traefik to create a `null` TXT record.
> 
> The support will continue on https://community.containo.us/t/unable-to-obtain-acme-certificate-for-wildcard-domains/5408

OK @ldez thanks for pointing this out.

**Comment by floatingpurr** [comment]
As recommended, I opened an issue for lego here: https://github.com/go-acme/lego/issues/1113

Thanks

**Comment by traefiker** [comment]
Closed by #6727.
